packages/build/src/plugins_core/functions/index.ts

```diff
@@ -248,31 +248,17 @@ const hasFunctionsDirectories = async function ({
   return false
 }
 
-// Takes a list of return values and produces an array with the paths of all
-// generated functions.
-const getGeneratedFunctionPaths = (returnValues: Record<string, ReturnValue>) => {
-  return Object.values(returnValues).flatMap(
-    (returnValue) => returnValue.generatedFunctions?.map((func) => func.path) || [],
-  )
-}
-
 // Takes a list of return values and produces an object with the names of the
 // generated functions for each generator. This is used for printing logs only.
-const getGeneratedFunctionsByGenerator = (returnValues: Record<string, ReturnValue>) => {
-  const result: Record<string, { displayName: string; generatorType: string; functionNames: string[] }> = {}
-
-  for (const id in returnValues) {
-    const { displayName, generatedFunctions, generatorType } = returnValues[id]
-
-    if (!generatedFunctions || generatedFunctions.length === 0) {
-      continue
-    }
-
-    result[id] = {
-      displayName,
-      generatorType,
-      functionNames: generatedFunctions.map((func) => basename(func.path)),
-    }
+const getGeneratedFunctionsByGenerator = (
+  generatedFunctions: GeneratedFunction[],
+): Record<string, GeneratedFunction[]> => {
+  const result: Record<string, GeneratedFunction[]> = {}
+
+  for (const func of generatedFunctions) {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    result[func.generator.name] = result[func.generator.name] || []
```

Comment:
FWIW this is because `f = Record<string, Foo>` with [`noUncheckedIndexAccess`](https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess) disabled results in the type `F`, not `F | undefined`. ESLint is being pedantically correct here (arguably).

You can avoid the suppression this way:
```suggestion
  const result: Record<string, GeneratedFunction[] | undefined> = {}

  for (const func of generatedFunctions) {
    result[func.generator.name] = result[func.generator.name] || []
```

Reply:
But that's not the type I want to return. I don't want consumers of this method to think that some keys will have the value `undefined`, because that shouldn't be the case.